### ISSUE TITLE
How to "ignore" a detection of a class in yolo loss function

### ISSUE THREAD

**Issue by kevinsu628** [qqwweee/keras-yolo3#545]
I am trying to modify the code to ignore some of the outputs that yolo predicts, but I am not sure how. 

For example, during training, the model predicts a car and a bike in an image. How can I modify the loss function, so that the bike prediction is neglected but the car prediction is kept? 

The reason I am doing this is, I want to combine different datasets that have different labels. For example, dataset_A labels bikes but not cars, dataset_B labels both bikes and cars. I want the training ignores its prediction of cars in dataset_A so that it won't affect the back prop during training. 

My idea is to generate a vector that tells if a dataset labels what class. (e.g. [1, 0, 1, 1] means the dataset doesn't label the second class, but every other classes). Then I need to multiply this vector by the inference output in the loss function so that it's the same as not detecting anything. 

If someone can point out if this is possible in yolo, or tell me where I should look into, I'd appreciate it.

**Comment by a793719595** [comment]
i don not think you should ignore those predictions, it can alse pass the gradient to make your model better.

**Comment by ThomasObot** [comment]
@kevinsu628 I'm interested to hear if you've been able to achieve decent results with this strategy. I would like to implement this too, but I have strong doubts as to how realistic it is to expect decent results.

**Comment by wowo68** [comment]
这是来自QQ邮箱的假期自动回复邮件。你好，我最近正在休假中，无法亲自回复你的邮件。我将在假期结束后，尽快给你回复。
